tests/test_synced_collection.py

```diff
@@ -476,39 +499,42 @@ def test_reset(self, synced_list):
         with pytest.raises(ValueError):
             synced_list.reset(1)
 
+    @given(JSONDataStrategy)
     def test_insert(self, synced_list, testdata):
         synced_list.reset([1, 2])
         assert len(synced_list) == 2
         synced_list.insert(1, testdata)
         assert len(synced_list) == 3
         assert synced_list[1] == testdata
 
-    def test_reversed(self,  synced_list):
-        data = [1, 2, 3]
-        synced_list.reset([1, 2, 3])
-        assert len(synced_list) == 3
+    @given(st.lists(JSONDataStrategy, max_size=5))
+    def test_reversed(self,  synced_list, data):
+        synced_list.reset(data)
+        assert len(synced_list) == len(data)
         assert synced_list == data
         for i, j in zip(reversed(synced_list), reversed(data)):
             assert i == j
 
-    def test_remove(self, synced_list):
-        synced_list.reset([1, 2])
+    @given(d1=JSONDataStrategy, d2=JSONDataStrategy)
+    def test_remove(self, synced_list, d1, d2):
+        synced_list.reset([d1, d2])
         assert len(synced_list) == 2
-        synced_list.remove(1)
+        synced_list.remove(d1)
         assert len(synced_list) == 1
-        assert synced_list[0] == 2
-        synced_list.reset([1, 2, 1])
-        synced_list.remove(1)
+        assert synced_list[0] == d2
+        synced_list.reset([d1, d2, d1])
+        synced_list.remove(d1)
         assert len(synced_list) == 2
-        assert synced_list[0] == 2
-        assert synced_list[1] == 1
+        assert synced_list[0] == d2
+        assert synced_list[1] == d1
```

Comment:
We should add a test for removing an element that is not at the front of the list. For example, removing `2` from `[1, 2, 1]` should leave `[1, 1]`. This would be easy, just copy and alter the second half of this test method (highlighted here). You can leave it as part of `test_remove`, just put it at the end below this chunk of code.